From 5b99262b1f3bca7b090659ba93129598b0a3d281 Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Tue, 17 Jun 2025 12:24:05 +0530
Subject: [PATCH] fix: pin react-icons version to fix TypeScript errors

TypeScript errors were occurring in jaeger-ui/plexus MiniMap.tsx where Heroicons components
(HiMagnifyingGlassPlus, HiMagnifyingGlassMinus, HiMiniArrowsPointingOut)
couldn't be used as JSX components. The error indicates that the IconType
isn't assignable to a valid JSX element type.

Pinning react-icons version(removing the caret ^) fixes
this issue, This is a known issue in react-icons (issue #1006) where the IconType interface in
newer versions is incompatible with direct JSX usage.

References:
- Issue: https://github.com/react-icons/react-icons/issues/1006
- Solution: https://github.com/react-icons/react-icons/issues/1006\#issuecomment-2704673971

This can be removed in the future when upstream updates the types to be compatible with JSX. Try to build without this patch if updateam update react-icons version

Signed-off-by: Debasish Biswas <debasishbsws.dev@gmail.com>
---
 packages/plexus/package.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/plexus/package.json b/packages/plexus/package.json
index c49bd0f3..840ab218 100644
--- a/packages/plexus/package.json
+++ b/packages/plexus/package.json
@@ -53,7 +53,7 @@
     "d3-selection": "^3.0.0",
     "d3-zoom": "^3.0.0",
     "memoize-one": "6.0.0",
-    "react-icons": "^5.0.1"
+    "react-icons": "5.0.1"
   },
   "scripts": {
     "_tasks/build/lib/js": "babel src --extensions '.tsx,.js' --out-dir lib",
--
2.39.5 (Apple Git-154)
